OneSignalSDK/unittest/src/test/java/com/test/onesignal/InAppMessageIntegrationTests.java

```diff
@@ -633,6 +635,160 @@ public void testInAppMessageClickActionOutcomeWithValue() throws Exception {
         assertEquals(1, iamOutcomeRequest.payload.get("device_type"));
     }
 
+    @Test
+    public void testOnMessageActionOccurredOnMessageSendOutcomeV2() throws Exception {
+        // Enable IAM v2
+        preferences = new MockOSSharedPreferences();
+        preferences.saveBool(preferences.getPreferencesName(), preferences.getOutcomesV2KeyName(), true);
+        trackerFactory = new OSTrackerFactory(preferences, new MockOSLog());
+        sessionManager = new MockSessionManager(OneSignal_getSessionListener(), trackerFactory, new MockOSLog());
+
+        OneSignal_setSharedPreferences(preferences);
+        OneSignal_setTrackerFactory(trackerFactory);
+        OneSignal_setSessionManager(sessionManager);
+
+        final OSTestInAppMessage message = InAppMessagingHelpers.buildTestMessageWithSingleTrigger(
+                OSTriggerKind.CUSTOM, "test_1", OSTestTrigger.OSTriggerOperator.EQUAL_TO.toString(), 2);
+
+        setMockRegistrationResponseWithMessages(new ArrayList<OSTestInAppMessage>() {{
+            add(message);
+        }});
+
+        // 1. Init OneSignal
+        OneSignalInit();
+        threadAndTaskWait();
+
+        // Enable influence outcomes
+        trackerFactory.saveInfluenceParams(new OneSignalPackagePrivateHelper.RemoteOutcomeParams());
+
+        final OSInAppMessageAction[] lastAction = new OSInAppMessageAction[1];
+        OneSignal.getCurrentOrNewInitBuilder().setInAppMessageClickHandler(new OneSignal.InAppMessageClickHandler() {
+            @Override
+            public void inAppMessageClicked(OSInAppMessageAction result) {
+                lastAction[0] = result;
+                // Ensure we are on the main thread when running the callback, since the app developer
+                //   will most likely need to update UI.
+                assertMainThread();
+
+                OneSignal.sendOutcome("test");
+                // Ensure outcome is sent
+                ShadowOneSignalRestClient.Request iamOutcomeRequest = ShadowOneSignalRestClient.requests.get(4);
```

Comment:
Below you are doing a lot of `Request` asserts
We should continuously update `RestClientAsserts.java` whenever we find new endpoints for testing
Ask me for help or check the current implementation and try to mimic it for the `measure_sources` endpoint